lib/resources/groups.rb

```diff
@@ -154,6 +158,35 @@ def groups
     end
   end
 
+  # OSX uses opendirectory for groups, so `/etc/group` may not be accurate
+  class DarwinGroup < GroupInfo
+    def groups
+      raw_output = inspec.command('dscacheutil -q group').stdout
+      raw_group_info = raw_output.split("\n\n")
+
+      group_info = []
+      regex = /^([^:]*?)\s*:\s(.*?)\s*$/
+      raw_group_info.each do |data|
+        group_info << inspec.parse_config(data, assignment_regex: regex).params
+      end
+
+      match_etc_group_output(group_info)
+    end
+
+    # Converts the `dscacheutil` groups to match `inspec.etc_group.entries`
+    def match_etc_group_output(dscacheutil_groups)
+      dscacheutil_groups.each { |g| g['gid'] = g['gid'].to_i }
+
+      # Convert `users` to `members` and ` ` to `,`
+      dscacheutil_groups.each do |g|
+        next if g['users'].nil?
+        g['members'] = g['users']
```

Comment:
The `#delete` method returns the content of the key that was deleted. You should be able to consolidate this line and the line after to:

```ruby
g['members'] = g.delete('users')
```

Reply:
Oh neat! Great tip. That's much cleaner.